base/compiler/ssair/driver.jl

```diff
@@ -118,24 +152,37 @@ function slot2reg(ir::IRCode, ci::CodeInfo, nargs::Int, sv::OptimizationState)
     return ir
 end
 
-function run_passes(ci::CodeInfo, nargs::Int, sv::OptimizationState)
-    preserve_coverage = coverage_enabled(sv.mod)
-    ir = convert_to_ircode(ci, copy_exprargs(ci.code), preserve_coverage, nargs, sv)
+function compact_all!(ir::IRCode)
+    length(ir.stmts) == 0 && return ir
```

Comment:
style nit:
```suggestion
    isempty(ir.stmts) && return ir
```
Is this actually legal IR? I though it had to contain `:return`, at least